aws-predictions/src/main/java/com/amplifyframework/predictions/aws/configuration/IdentifyEntitiesConfiguration.java

```diff
@@ -0,0 +1,136 @@
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.predictions.aws.configuration;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.amplifyframework.predictions.aws.NetworkPolicy;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * Configures the behavior for text interpretation.
+ */
+public final class IdentifyEntitiesConfiguration {
+
+    private static final int MAX_VALID_ENTITIES = 50;
+
+    private final int maxEntities;
+    private final String collectionId;
+    private final boolean generalEntityDetection;
+    private final boolean celebrityDetectionEnabled;
+    private final NetworkPolicy networkPolicy;
+
+    private IdentifyEntitiesConfiguration(
+            int maxEntities,
+            String collectionId,
+            boolean generalEntityDetection,
+            boolean celebrityDetectionEnabled,
+            NetworkPolicy networkPolicy
+    ) {
+        this.maxEntities = maxEntities;
+        this.collectionId = collectionId;
+        this.generalEntityDetection = generalEntityDetection;
+        this.celebrityDetectionEnabled = celebrityDetectionEnabled;
+        this.networkPolicy = networkPolicy;
+    }
+
+    /**
+     * Construct an instance of {@link IdentifyEntitiesConfiguration} from
+     * plugin configuration JSON object.
+     * @param configurationJson the plugin configuration
+     * @return the configuration for entities detection
+     * @throws JSONException if identify configuration is malformed
+     */
+    @Nullable
+    public static IdentifyEntitiesConfiguration fromJson(@NonNull JSONObject configurationJson) throws JSONException {
+        if (!configurationJson.has("identifyEntities")) {
+            return null;
+        }
+
+        // Required fields
+        JSONObject identifyEntitiesJson = configurationJson.getJSONObject("identifyEntities");
+        String celebEnabledString = identifyEntitiesJson.getString("celebrityDetectionEnabled");
+        String networkPolicyString = identifyEntitiesJson.getString("defaultNetworkPolicy");
+
+        boolean celebEnabled = Boolean.parseBoolean(celebEnabledString);
+        NetworkPolicy networkPolicy = NetworkPolicy.fromKey(networkPolicyString);
+
+        // Optional fields
+        int maxEntities;
+        boolean generalEntity;
+        String collectionId;
+        String maxEntitiesString;
+        try {
+            maxEntitiesString = identifyEntitiesJson.getString("maxEntities");
+            collectionId = identifyEntitiesJson.getString("collectionId");
```

Comment:
So if this throws, you'lll end up in the `catch` block, and then `maxEntities` will be `0`, and `generalEntity` will be `true`. Is that the control flow you were going for, here? Seems like there's a lot of branches here, depending on which lines throws, and what the defaults will get set to.

Actually, I guess if `maxEntitiesString` line throws, then the outcome here is even weirder, with `maxEntitiesString` as `null` and `collectionId` as `""`.

`generalEntity` is very hard to trace. It starts as `false` (default for a boolean.) It _might_ get set to true/false on L83. But. But if there is an error before L83, it gets reset to `true` in the `catch`.

Reply:
https://docs.amplify.aws/lib/predictions/identify-entity?platform=ios

> The value of maxEntities should be a number greater than 0 or less than 51 (50 is the max number of entities Rekognition can detect from a collection). If both collectionId and maxEntities do not have valid values in the amplifyconfiguration.json file, then this call will just detect entities in general with facial features, landmarks, etc.

Very weird flow indeed, but I think what I currently have kind of describes this sufficiently?

Reply:
also `generalEntity` (which is pretty badly named I guess. Should have been something like `isGeneralEntityDetectionEnabled`, but that was too long haha) should kind of be defaulted to `true`. However, the IDE saw that its value is specified in both try and catch clauses, so it didn't like being given n initial value.

The main idea is, `isGeneralEntityDetection` should be true, and `collectionId` and `maxEntities` should not be referenced if it is. Therefore, they are given arbitrary values for the time being (inside `catch`). `isGeneralEntityDetection` will be true ONLY if both `maxEntities` and `collectionId` are specified in the configuration, and the `maxEntities` value is valid (0 < x < 51).